examples/ebpf-allowed-memory.rs

```diff
@@ -0,0 +1,40 @@
+#![no_std]
+#![no_main]
+
+use aya_ebpf::{
+    bindings::{BPF_F_NO_PREALLOC, TC_ACT_PIPE},
+    macros::{classifier, map},
+    maps::HashMap,
+    programs::TcContext,
+};
+
+#[no_mangle]
+#[link_section = "license"]
+pub static LICENSE: [u8; 13] = *b"Dual MIT/GPL\0";
```

Comment:
We don't have to use GPL in our case, and I'd rather avoid including GPL-licensed code (or even examples) in the repo. Would you be ready to use a different license for these?

We probably don't need to specify the license in a dedicated section by the way, rbpf does not take it into account. However, I'd like to have a SPDX header on the new files, please.

Reply:
Any code that wants to be loaded in the kernel will need it though, so I included it because someone will copy/paste :-) The cilium people wrote a nice blog article about it.
It is dual licensed to MIT.

I don't feel strongly about this, so I'll just remove those lines.

Reply:
> Any code that wants to be loaded in the kernel will need it though, so I included it because someone will copy/paste :-)

Sorry, but this is incorrect :) The restriction applies to some program types and some features, such as some of the available helpers. Basically, everything that interacts with kernel internals needs to be GPL to be loaded. As for TC classifiers and the map lookup helper, there's no such restriction, so you could load it even if it's not GPL.

> The cilium people wrote a nice blog article about it. It is dual licensed to MIT.

I'm a Cilium person myself :) They were coworkers at the time. Yes, the dual license helps, but here there's no need to introduce GPL at all in the repo, so I'd just avoid it as long as I can.

> I don't feel strongly about this, so I'll just remove those lines.

Thank you! :pray: